File: easy/1716_calculate_money_in_leetcode_bank/solution.py
class Solution:
    def totalMoney(self, n: int) -> int:
        weeks, days = divmod(n, 7)
        return (
            weeks * 28
            + 7 * (weeks - 1) * weeks // 2
            + days * (2 * weeks + days + 1) // 2
        )

File: easy/1716_calculate_money_in_leetcode_bank/solution.cpp
using namespace std;

class Solution {
public:
  int totalMoney(int n) {
    int weeks = (int)n / 7, days = n % 7;
    return weeks * 28 + 7 * (weeks - 1) * weeks / 2 +
           days * (2 * weeks + days + 1) / 2;
  }
};
